src/plugins/intel_cpu/src/emitters/jit_eltwise_emitters.cpp

```diff
@@ -1958,8 +1959,194 @@ void jit_soft_sign_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, con
+        h->vpbroadcastd(vmm_dst | ones_mask | h->T_z, reg32_one);
+    } else {
+        h->uni_vxorps(vmm_dst, vmm_dst, vmm_dst);
+    }
+}
+
+template <x64::cpu_isa_t isa>
+void jit_is_inf_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs) const {
+    using Vmm = typename conditional<isa == x64::sse41, Xmm, Ymm>::type;
+
+    if (detect_negative || detect_positive) {
+        auto vmm_src = Vmm(in_vec_idxs[0]);
+        auto vmm_dst = Vmm(out_vec_idxs[0]);
+
+        if (detect_positive) {
+            if (detect_negative) {
+                h->uni_vpslld(vmm_src, vmm_src, 1);
+                h->uni_vpsrld(vmm_src, vmm_src, 1);
+            }
+            h->uni_vpcmpeqd(vmm_src, vmm_src, table_val("inf"));
+        } else if (detect_negative) {
+            h->uni_vpcmpeqd(vmm_src, vmm_src, table_val("inf_neg"));
+        }
+
+        if (isa == x64::avx2) {
+            h->uni_vandps(vmm_dst, vmm_src, table_val("one"));
+        } else {
+            if (vmm_src.getIdx() != vmm_dst.getIdx()) {
+                h->uni_vmovups(vmm_dst, vmm_src);
+            }
+            h->uni_vandps(vmm_dst, vmm_dst, table_val("one"));
+        }
+    } else {
+        auto vmm_dst = Vmm(out_vec_idxs[0]);
+        h->uni_vxorps(vmm_dst, vmm_dst, vmm_dst);
+    }
+}
+
+void jit_is_inf_emitter::emit_impl(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs,
+                                   const std::vector<size_t> &pool_vec_idxs, const std::vector<size_t> &pool_gpr_idxs,
+                                   const emitter_context *emit_context) const {
+    if (host_isa_ == x64::avx512_core) {
+        emit_isa<x64::avx512_core>(in_vec_idxs, out_vec_idxs);
+    } else if (host_isa_ == x64::avx2) {
+        emit_isa<x64::avx2>(in_vec_idxs, out_vec_idxs);
+    } else if (host_isa_ == x64::sse41) {
+        emit_isa<x64::sse41>(in_vec_idxs, out_vec_idxs);
+    } else {
+        IE_THROW() << "jit_is_inf_emitter doesn't support ISA " << host_isa_;
+    }
+}
+
+void jit_is_inf_emitter::register_table_entries() {
+    if (host_isa_ != x64::avx512_core) {
+        push_arg_entry_of("one", CONST_1_F, true);
+        push_arg_entry_of("inf", INF_MASK, true);
+        push_arg_entry_of("inf_neg", INF_NEG_MASK, true);
+    }
+}
+
+/// IS_NAN ///
+template <>
+void jit_is_nan_emitter::emit_isa<x64::avx512_core>(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs) const {
+    auto vmm_src = Zmm(in_vec_idxs[0]);
+    auto vmm_dst = Zmm(out_vec_idxs[0]);
+    auto &ones_mask = h->k1;
+    auto reg32_one = Reg32(aux_gpr_idxs[0]);
+
+    h->mov(reg32_one, CONST_1_F);
+    h->vfpclassps(ones_mask, vmm_src, 0B10000001);
+    h->vpbroadcastd(vmm_dst | ones_mask | h->T_z, reg32_one);
+}
+
+template <x64::cpu_isa_t isa>
+void jit_is_nan_emitter::emit_isa(const std::vector<size_t> &in_vec_idxs, const std::vector<size_t> &out_vec_idxs) const {
```

Comment:
in scalar code to check if the value is NaN it is enough to compare it with itself: res = val != val (return true only if val is Nan). I am eondering can we achieve the same with vcmps?

Reply:
cmpps works as well. Modified.